### ISSUE TITLE
[java] New rule, UnnecessaryInitCause

### ISSUE THREAD

**Issue by aaime** [pmd/pmd#3821]
**UnnecessaryInitCause:**

**Proposed Category:** Best Practices

**Description:** Legacy code can be using ``initCause`` to track the cause of an exception, while it's now possible to provide the same information via a constructor argument. For example, ``IOException`` got constructors supporting the specification of a cause since Java 1.6, code written earlier is likely to use ``initCause`` instead.

**Code Sample:** This should include code, that should be flagged by the rule. If possible, the "correct" code
according to this new rule should also be demonstrated.

```
// issue example
... catch(MyException exception) {
  IOException exception = new IOException("Something bad happened");
  exception.initCause(exception);
  throw exception;
}
// better approach
... catch(MyException exception) {
  throw new IOException("Something bad happened", exception);
}
```

In order to keep it general (rather than focusing on IOException) I'm guessing the rule should be written in Java, to verify there is another constructor taking the same arguments, plus the cause (as the last argument).

**Comment by oowekyala** [comment]
Interesting idea, thanks for the suggestion!
